full/Dockerfile

```diff
@@ -1,192 +1,149 @@
+    && apt-get install -yq \
+        clang-format-6.0 \
+        clang-tools-6.0 \
+        cmake \
+    && ln -s /usr/bin/clangd-6.0 /usr/bin/clangd \
+    && apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/*
 
-RUN echo oracle-java8-installer shared/accepted-oracle-license-v1-1 select true | /usr/bin/debconf-set-selections
-RUN add-apt-repository ppa:webupd8team/java \
-  && apt-get update \
-  && apt-get install -y \
-        oracle-java8-installer \
+### Java & Maven ###
+RUN add-apt-repository -yu ppa:webupd8team/java \
+    && echo oracle-java8-installer shared/accepted-oracle-license-v1-1 select true | debconf-set-selections \
+    && apt-get install -yq \
         gradle \
-  && apt-get clean && rm -rf /var/cache/apt/* && rm -rf /var/lib/apt/lists/* && rm -rf /tmp/*
+        oracle-java8-installer \
+    && apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/*
 
-# Maven
 ARG MAVEN_VERSION=3.5.4
-ARG USER_HOME_DIR="$HOME"
-ARG SHA=ce50b1c91364cb77efe3776f756a6d92b76d9038b0a0782f7d53acf1e997a14d
-ARG BASE_URL=https://apache.osuosl.org/maven/maven-3/${MAVEN_VERSION}/binaries
-
-RUN mkdir -p /usr/share/maven /usr/share/maven/ref \
-  && curl -fsSL -o /tmp/apache-maven.tar.gz ${BASE_URL}/apache-maven-${MAVEN_VERSION}-bin.tar.gz \
-  && echo "${SHA}  /tmp/apache-maven.tar.gz" | sha256sum -c - \
-  && tar -xzf /tmp/apache-maven.tar.gz -C /usr/share/maven --strip-components=1 \
-  && rm -f /tmp/apache-maven.tar.gz \
-  && ln -s /usr/share/maven/bin/mvn /usr/bin/mvn
-ENV MAVEN_HOME /usr/share/maven
-ENV MAVEN_CONFIG "$USER_HOME_DIR/.m2"
-ENV PATH $PATH:${MAVEN_HOME}/bin
-
-#C/C++
-RUN curl https://apt.llvm.org/llvm-snapshot.gpg.key | apt-key add -
-RUN echo "deb http://apt.llvm.org/bionic/ llvm-toolchain-bionic-6.0 main" > /etc/apt/sources.list.d/llvm.list
-RUN apt-get update \
-    && apt-get install -y \
-        clang-tools-6.0 \
-        cmake \
-        clang-format \
-    && apt-get clean && rm -rf /var/cache/apt/* && rm -rf /var/lib/apt/lists/* && rm -rf /tmp/*
-RUN ln -s /usr/bin/clangd-6.0 /usr/bin/clangd
-
-# PHP
-ENV DEBIAN_FRONTEND=noninteractive
-RUN apt-get update \
-    && apt-get install -y \
-        php \
-        apache2 \
+ENV MAVEN_HOME=/usr/share/maven \
+    MAVEN_CONFIG=$HOME/.m2
+ENV PATH=$MAVEN_HOME/bin:$PATH
+RUN mkdir -p $MAVEN_HOME \
+    && curl -fsSL https://apache.osuosl.org/maven/maven-3/$MAVEN_VERSION/binaries/apache-maven-$MAVEN_VERSION-bin.tar.gz \
+        | tar -xzvC $MAVEN_HOME --strip-components=1
+
+### PHP ###
+RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -yq \
         composer \
+        php \
         php-all-dev \
-        php-intl \
-        php-mysql \
-        php-pgsql \
-        php-xml \
+        php-curl \
+        php-date \
         php-gd \
+        php-gettext \
+        php-intl \
         php-json \
-        php-curl \
         php-mbstring \
+        php-mysql \
+        php-net-ftp \
+        php-pgsql \
```

Comment:
php-xml and php-ctype are not listed anymore... is this intentional?

Reply:
I added everything I needed to get the https://github.com/symfony/demo running and I think I also added some extensions needed by Laravel. 
I'd prefer to explicitly add all  packages that a user would expect. As you said, dependencies may change over time. It's also possible that at some point we intentionally decide to remove one extension. This should not cause other extensions to disappear.

Reply:
OK I'll put it back as it was there ...